app/src/main/java/org/sagebase/crf/step/active/HeartBeatDetector.java

```diff
@@ -150,17 +156,20 @@ void fillHsv(HeartBeatSample sample) {
         float max = Math.max(sample.r, Math.max(sample.g, sample.b));
         float delta = max - min;
 
-        if (Math.round(delta * 1000.0f) == 0) {
+        if (((int)Math.round(delta * 1000.0) == 0) || ((int)Math.round(delta * 1000.0) == 0)) {
+            sample.h = -1;
+            sample.s = 0;
+            sample.v = 0;
             return;
         }
 
         float hue;
         if (sample.r == max) {
-            hue = (float)(sample.g - sample.b) / delta;
+            hue = (sample.g - sample.b) / delta;
```

Comment:
I don't understand these formulas? Do you have a link I could read up on?

Reply:
Direct copy from the iOS code.  Also, not really too important, since hsv can easily be calculated by the server based on r, g, b just like we are doing here.